### PR TITLE
feat: Silder: adjust fillStart calculation for value=0 and normalization function

### PR DESCRIPTION
**Pull request by spdev3000**
This PR adds a way to use fill-start="0" as reference point as well as respecting a provided normalization function to correctly calculate fill offset and length.

## Description

First check, if we really just have the attribute `fill-start` provided with no number value, if so, use centered point of track like before.
For all other number values, use normalization function of model of active handle to calculate real fill-offset and fill-length.

## Related issue(s)

#4559 

## Motivation and context

2 current problems:
1. Currently fill-start attribute only allows to set values !== 0, but sometimes clients have sliders with min<0 , so for example: min="-100", max="200" - so we need a possibility to define fill-start="0" to not start from centered track position
2. Also currently the fill-start calculation doesn't respect any provided normalization function, so we'd like to calculate all fill-related values (offset + length) by a given normalization function of model of active handle

## How has this been tested?

2 new test cases have been added into slider.stories.ts as well as into slider.test.ts 

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/slider/src/Slider.ts [hunk 1]
```diff
@@ -545,12 +546,13 @@ export class Slider extends SizedMixin(ObserveSlotText(SliderHandle, ''), {
     protected override willUpdate(changed: PropertyValues): void {
         if (changed.has('value') && changed.has('fillStart')) {
             this._cachedValue = Number(this.value);
-            if (this.fillStart) {
-                this.centerPoint = Number(this.fillStart);
-            } else {
+            // Test if fill-start is set without a value
+            if (this.getAttribute('fill-start') === '') {
```
Comment (spdev3000) [inline packages/slider/src/Slider.ts:550]: TBH I don't see any other option to check, if `fill-start` is provided, but without any value. As soon as I check on the property `fillStart` it converts both empty string and `0` to the number `0`, where I cannot decide, was it empty or really the value `0`.
Any ideas to improve?

Comment (Rajdeepc) [reply packages/slider/src/Slider.ts:550]: How about you check for the type of `fillStart`? It would have a value if it is not type of a `boolean`

Comment (spdev3000) [reply packages/slider/src/Slider.ts:550]: `typeof this.fillStart` always returns `number` and not `boolean` as lit maps that attribute always to a number property.

**Comment by Rajdeepc** [comment]
@spdev3000 Can you also add some documentation around it! That would be very helpful to let the consumers know about the flexibility to use the min-max ranges from -ve minimum values.

**Comment by spdev3000** [comment]
will be continued as #4573

**Pull request closed without merging**